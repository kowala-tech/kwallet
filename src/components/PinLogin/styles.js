// const toolbarHeight = 75;

const styles = (theme) => ({
	flexContainer: {
		display: "flex",
		flexDirection: "column",
		alignItems: "center",
		justifyContent: "center",
		textAlign: "center"
	},
	lockIcon: {
		marginBottom: theme.spacing.unit * 6
	},
	unlockMessage: {
		marginTop: theme.spacing.unit * 6,
		textAlign: "center",
		color: theme.palette.primary.contrastText,
		fontWeight: 200
	},
	close: {
		width: theme.spacing.unit * 4,
		height: theme.spacing.unit * 4,
	},
});

export default styles;
