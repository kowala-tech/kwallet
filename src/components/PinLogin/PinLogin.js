import React from "react";
import PropTypes from "prop-types";
import { compose } from "recompose";
import classNames from "classnames";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import LockOutline from "@material-ui/icons/LockOutline";
import Grow from "@material-ui/core/Grow";
import Typography from "@material-ui/core/Typography";
// Component Related Imports
import styles from "./styles";
import PinForm from "./components/PinForm";

class PinLogin extends React.Component {

	render() {
		const {
			classes,
			handleSubmit,
			error,
			//errorMessage,
			loading,
			username
		} = this.props;

		return (
			<div>
				<Grow in>
					<div className={classes.flexContainer}>
						<LockOutline className={classNames("kowala",[classes.lockIcon])}/>
						<PinForm
							onSubmit={handleSubmit}
							loading={loading}
							error={error}
							username={username}
						/>
						<Typography variant="body1"
							className={classes.unlockMessage}>
							Enter your PIN to unlock your wallet
						</Typography>
					</div>
				</Grow>
			</div>
		);
	}
}

PinLogin.propTypes = {
	classes: PropTypes.object.isRequired,
	handleSubmit: PropTypes.func.isRequired,
	loading: PropTypes.bool.isRequired,
	error: PropTypes.bool.isRequired,
	errorMessage: PropTypes.string,
	username: PropTypes.string.isRequired
};

export default compose(
	withStyles(styles),
)(PinLogin);
