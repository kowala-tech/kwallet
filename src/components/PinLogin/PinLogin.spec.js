import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import PinLogin from "./PinLogin";

describe("PinLogin", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<PinLogin
					handleSubmit={jest.fn()}
					loading={false}
					error={false}
					errorMessage={""}
					username={"test-user"}
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
