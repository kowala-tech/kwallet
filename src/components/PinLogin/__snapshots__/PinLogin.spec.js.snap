// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinLogin it should match the snapshot 1`] = `
<PinLogin
  classes={
    Object {
      "close": "PinLogin-close-5",
      "flexContainer": "PinLogin-flexContainer-1",
      "lockIcon": "PinLogin-lockIcon-2",
      "text": "PinLogin-text-4",
      "unlockMessage": "PinLogin-unlockMessage-3",
    }
  }
  error={false}
  errorMessage=""
  handleSubmit={[MockFunction]}
  loading={false}
  username="test-user"
/>
`;
